File: src/assets/code-examples/home/yepcode-run.js
const { YepCodeRun } = require("@yepcode/run");

const runner = new YepCodeRun({ apiToken: "<your-api-token-here>" });

// Execute any JavaScript or Python code (you may use dependencies)
const execution = await runner.run(
  `async function main() {
    const message = "Hello, YepCode!";
    console.log(message);
    return { message };
  }
  module.exports = { main };`,
  {
    onLog: (log) =>
      console.log(`${log.timestamp} ${log.level}: ${log.message}`),
    onFinish: (returnValue) => console.log("Finished:", returnValue),
    onError: (error) => console.error("Error:", error),
  }
);

// Wait for execution to complete
await execution.waitForDone();

// Retrieve an existing execution
const existingExecution = await runner.getExecution("execution-id");

File: src/assets/home/code-examples/yepcode-run.py
from yepcode_run import YepCodeRun, YepCodeApiConfig

runner = YepCodeRun(YepCodeApiConfig(api_token="<your-api-token-here>"))

# Execute any JavaScript or Python code (you may use dependencies)
execution = runner.run(
    """def main():
    message = "Hello, YepCode!"
    print(message)
    return {"message": message}""",
    {
        "onLog": lambda log: print(f"{log.timestamp} {log.level}: {log.message}"),
        "onFinish": lambda return_value: print("Finished:", return_value),
        "onError": lambda error: print("Error:", error),
    },
)

# Wait for execution to complete
execution.wait_for_done()

# Retrieve an existing execution
existing_execution = runner.get_execution("execution-id")
